cmake/third_party/gcp_sdk.cmake

```diff
@@ -0,0 +1,75 @@
+cmake_minimum_required(VERSION 3.10)
+include(ExternalProject)
+include(FetchContent)
+include(GNUInstallDirs)
+include(${CMAKE_SOURCE_DIR}/cmake/helpers.cmake)
+
+# Skip the GCP SDK build step if the extension is not enabled.
+if(NOT ENABLE_GCP)
+    return()
+endif()
+
+config_choice(
+    IMPORT_GCP_SDK
+    "Specify how to import the GCP SDK"
+    OPTIONS
+        "none;IMPORT_GCP_SDK_NONE;NOT ENABLE_GCP"
+        "package;IMPORT_GCP_SDK_PACKAGE;ENABLE_GCP"
+        "external;IMPORT_GCP_SDK_EXTERNAL;ENABLE_GCP"
+)
+
+if(IMPORT_GCP_SDK_NONE)
+    message(FATAL_ERROR "Cannot enable GCP extension without specifying an IMPORT_GCP_SDK method (package, external).")
+endif()
+
+if(IMPORT_GCP_SDK_PACKAGE)
+    message(FATAL_ERROR "Currently the package config option is not implement yet.")
```

Comment:
nit: This reads a bit clunky. Can we change it to something like `Importing GCP via package is not currently implemented`

Reply:
changed: 03c2e3e3327853ae3c3662157e1744ad403aae83